packages/botonic-react/src/webchat-app.jsx

```diff
@@ -47,6 +52,36 @@ export class WebchatApp {
     this.visibility = visibility
     this.webchatRef = createRef()
     this.appId = appId
+    this.shouldRender = false
+    this.shouldRenderOptionsAtRuntime = {}
+
+    onDOMLoaded(() => {
+      this.createRootElement()
+      if (this.shouldRender) {
+        this.render(this.getReactMountNode(), this.shouldRenderOptionsAtRuntime)
+        this.shouldRender = false
+        this.shouldRenderOptionsAtRuntime = {}
+      }
+    })
+  }
+
+  createRootElement() {
+    // Create root element <div id='root'> if not exists
+    // Create shadowDOM to root element if needed
+    let host = document.getElementById(this.hostId)
+    if (!host) {
+      host = document.createElement('div')
+      host.id = this.hostId
+      if (document.body.firstChild)
+        document.body.insertBefore(host, document.body.firstChild)
+      else document.body.appendChild(host)
```

Comment:
Why inserting the root element as the first child? Why not doing only `document.body.appendChild(host)`?

Reply:
It's just a cosmetic thing: to put the root tag at the top of the body, so it's easy to find.